crates/ruff_python_semantic/src/model.rs

```diff
@@ -856,6 +865,27 @@ impl<'a> SemanticModel<'a> {
         &self.scopes[self.scope_id]
     }
 
+    /// Returns the parent of the given scope, if any.
+    pub fn scope_parent(&self, scope: &Scope) -> Option<&Scope<'a>> {
```

Comment:
Nit: It should be `parent_scope` to be consistent with `global_scope`. I also find it reads more naturally. It's the parent scope. Or you need to use an apostrophe: `scope's_parent`.

Reply:
Hmm I wrote it that way first but I was trying to be consistent with `expr_parent` / `expr_grandparent` / `stmt_parent`. The `global_scope` is a special kind of scope, not a scope relative to the current one.

Reply:
(The naming conventions on the model API are not very good and have evolved somewhat haphazardly.)

Reply:
I'll just change them for now. Not worth going back and forth I think. We can address it more later if needed.